deepinv/optim/optim_iterators/pgd.py

```diff
@@ -153,3 +154,101 @@ def forward(self, x, cur_prior, cur_params):
                 * cur_prior.grad(x, cur_params["g_param"])
             )
             return x - grad
+
+
+class PMDIteration(OptimIterator):
+    r"""
+    Iterator for Proximal Mirror Descent (PMD).
+
+    Class for a single iteration of the Proximal Mirror Descent (PMD) algorithm for minimizing :math:`f(x) + \lambda g(x)`.
+
+   For a given Bregman convex potential :math:`h`, the iteration is given by
+
+    .. math::
+        \begin{equation*}
+        \begin{aligned}
+        u_{k} &= \nabla h^*(\nabla h(x_k) - \gamma \nabla f(x_k)) \\
+        x_{k+1} &= \operatorname{prox^h}_{\gamma \lambda g}(u_k)
+        \end{aligned}
+        \end{equation*}
+
+
+   where :math:`\gamma` is a stepsize that should satisfy :math:`\gamma \leq 2/L` with :math:`L` verifying :math:`Lh-f` is convex.
+   The potential :math:`h` should be specified in the cur_params dictionary.
+
+    """
+
+    def __init__(self, bregman_potential=BregmanL2(), **kwargs):
+        super(PMDIteration, self).__init__(**kwargs)
+        self.bregman_potential = bregman_potential
+        self.g_step = gStepPGD(**kwargs)
+        self.f_step = fStepPGD(**kwargs)
+        if self.g_first:
+            self.requires_grad_g = True
+        else:
+            self.requires_prox_g = True
+
+
+class fStepPMD(fStep):
+    r"""
+    Proximal Mirror Descent fStep module.
+    """
+
+    def __init__(self, **kwargs):
+        super(fStepPGD, self).__init__(**kwargs)
+
+    def forward(self, x, cur_data_fidelity, cur_params, y, physics, bregman_potential):
```

Comment:
Should the bregman_potential be a parameter of the PMDIteration or an argument? It seems to both at the moment, but it seems more natural for me if it is a parameter

Reply:
yes it's a mistake here, let me correct it